### DISCUSSION TITLE
delete related records

### DISCUSSION THREAD

**Discussion by photojo** [filamentphp/filament#5025] [Help] [answered]
I have a table and a repeater with related records.

My model for 'users' ...

```
class User extends Model
{
....
 public function cars(): HasMany
    {
        return $this->hasMany(Cars::class, 'id_car', 'id_user');
    }
...
```
and in table 

  ```
...
Repeater::make('cars')->label('users cars')
                                ->relationship()
                                ->schema([
...
```

With

```
 DeleteAction::make('user delete')
                    ->modalHeading('delete user)
                    ->action(function (User $record): void {
                        $record->delete();
                      })
```
I can delete a user.

But how can I **also delete all cars**, which belongs to the user?!

**Comment by photojo** [comment]
public static function boot()
    {
        parent::boot();

        static::deleting(function ($user) {
            $user->cars->each(function ($item) {
                $item->delete();
            });
            // notifikation
        });
    }

**Comment by ArtMin96** [comment, marked as answer]
> DeleteAction::make('user delete')
>                     ->modalHeading('delete user)
>                     ->action(function (User $record): void {
>                         $record->delete();
>                       })

or

```php
DeleteAction::make('user delete')
    ->modalHeading('delete user)
    ->action(function (User $record): void {
        $record->cars()->each(function ($car) {
            $car->delete();
        });

        $record->delete();
    })
```

**Reply by photojo** [reply]
Thank you!
